frontend/src/metabase/public/components/widgets/EmbedModalContent.jsx

```diff
@@ -132,19 +131,30 @@ export default class EmbedModalContent extends Component {
     this.setState({ embeddingParams: resource.embedding_params || {} });
   };
 
-  getPreviewParams() {
+  getPreviewParameters(resourceParameters, embeddingParams) {
+    const lockedParameters = resourceParameters.filter(
+      parameter => embeddingParams[parameter.slug] === "locked",
+    );
+
+    return lockedParameters;
+  }
+
+  getPreviewParamsBySlug() {
     const { resourceParameters } = this.props;
     const { embeddingParams, parameterValues } = this.state;
-    const params = {};
-    for (const parameter of resourceParameters) {
-      if (embeddingParams[parameter.slug] === "locked") {
-        params[parameter.slug] =
-          parameter.id in parameterValues
-            ? parameterValues[parameter.id]
-            : null;
-      }
-    }
-    return params;
+
+    const lockedParameters = this.getPreviewParameters(
+      resourceParameters,
+      embeddingParams,
+    );
+
+    const parameterSlugValuePairs = lockedParameters.map(parameter => {
+      const value =
+        parameter.id in parameterValues ? parameterValues[parameter.id] : null;
+      return [parameter.slug, value];
+    });
+
+    return Object.fromEntries(parameterSlugValuePairs);
```

Comment:
this code, annoyingly, is kind of an exception to the rule in that it builds a slug/values map but it explicitly sets parameters that aren't found in `parameterValues` to `null`, meaning nil values are present in this slug map, unlike in other areas of the codebase where we filter out nil parameter values.

For now, I'm going to just leave this code as is, besides a bit of cleanup and renaming, so that I don't break something by changing the returned data structure.